appserver/connectors/work-management/src/main/java/com/sun/enterprise/connectors/work/WorkCoordinator.java

```diff
@@ -412,6 +419,7 @@ public String toString() {
      * @param state CREATED or Either STARTED or COMPLETED or TIMEDOUT
      */
     public synchronized void setState(int state) {
+        logger.finest(() -> String.format("setState(state=%s)", state));
```

Comment:
I would defend suppliers, because the code is optimized by JVM and I believe it is still faster than if block with JUL formatter; especially when you enable finest.
But we can argue about nanoseconds, the JUL logging is extremely slow by default because of poor implementation of handlers and formatters, concatenating strings, etc.
If I remember well, the lambda here is created pretty fast, because it is not generated in runtime b some other method, it uses some template and even JIT can inline it.

But for sure, I will create some test for that, because I don't believe too much what I wrote :)

Reply:
See test results I posted somewhere in the begining. It surprised even me - you can safely use lambdas where posiible, I'm pretty sure that if you don't use a method to generate new lambda, it is still the same.